guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc

```diff
@@ -133,13 +133,15 @@ ifdef::katello[]
 # {foreman-installer}
 ----
 endif::[]
-. Check when the kernel packages were last updated:
+. Check if the newest installed kernel is currently running:
 +
 [options="nowrap"]
 ----
-# rpm -qa --last | grep kernel
+# rpm --query --last kernel | head -n 1 | grep $(uname --kernel-release)
 ----
-. Optional: If a kernel update occurred since the last reboot, reboot the system:
+If this command produces the kernel version as output, it means that the newest installed kernel is running.
+Otherwise, the system needs a reboot to run the newest kernel.
+. If the previous command did not produce output, reboot the system:
 +
 ----
 # reboot
```

Comment:
What I struggle with is the balance. I mean that we are providing a somewhat complicated command that may fail. If it does fail, how does a user recover? What can they modify to make sure they do have the right info. Taking it to the extreme, you could suggest
```
rpm --query --last kernel | head -n 1 | grep $(uname --kernel-release) || reboot
```

Or even more extreme
```
rpm --query --last kernel | head -n 1 | grep $(uname --kernel-release) && echo "Running latest kernel" || reboot
```

But is the reader really helped by this? I'm not a technical writer so I don't know how to answer this, but my feeling is that it may be better to have simple commands.

At a certain point, it may be better to submit a feature request and document that feature.

Reply:
Perhaps you would prefer this version?
> 10. Check the newest installed kernel package:
> ```
> # rpm --query --last kernel | head -n 1
> ```
> 11. Check the currently running kernel package:
> ```
> # uname --kernel-release
> ```
> 12. Optional: If the versions of the newest installed and currently running kernel package differ, reboot the system:
> ```
> # reboot
> ```

Reply:
I think so yes, though I'd start with explaining why you're doing it first. Something like
> Optional: to determine if a reboot is needed, check the newest installed kernel package:
> ...
> Compare this to the running kernel version:
> ...
> If the versions ....

Reply:
Understood. See if you like the new version. I put determining whether system needs a reboot into one step with sub-steps, and then optional rebooting in another step.

Reply:
sub-steps are a good idea and something I also thought about.